ui/helpers/utils/token-util.js

```diff
@@ -154,7 +154,7 @@ export function getTokenAddressParam(tokenData = {}) {
  * @returns {string | undefined} A decimal string value.
  */
 export function getTokenValueParam(tokenData = {}) {
-  return tokenData?.args?._value?.toString();
+  return tokenData?.args?._value?.toString() || tokenData?.args?.[1];
```

Comment:
ah right. I was running into a test error when mocking out some token data for testing the edit send token transaction flow (which isn't hooked up but has a new test to cover it) and found that the getTokenValueParam method had no fall back of checking positional values, whereas the getTokenAddressParam did and they both operate on the same data structure. 

I think the fallback was added as part of the switch to ethers from whatever it was before? Either way if we have an existing fallback for operating on the same data structure in one method it should exist in the sister method as well, though its questionable if we need one of these lookups at all and should be factored out at some point when we're more sure of the data structure of this object.

Reply:
Apparently it was done differently intentionally. See this comment: https://github.com/MetaMask/metamask-extension/pull/9290#issuecomment-678579884
cc @rekmarks

Reply:
I think the key to understanding my change is [this diff](https://github.com/MetaMask/metamask-extension/commit/6279f39cb84fe0915e651a3453a6f316db233c3d#diff-406332500881d7c8502eed98ad6630e7046d74dd5836b95d03eb0256368e9806L141), from #9290. If you look at the [`human-standard-token-abi`](https://github.com/danfinlay/human-standard-token-abi/blob/master/index.js), you'll see that, except for the constructor, the first input of all ABI entities of `type: function` with any inputs is always an address. The second param is sometimes a `uint256` value, and sometimes and address. Therefore `getTokenValueParam` definitely can't fall back to the second positional parameter.

I have a lot of questions about the soundness of falling back to the first param in case there is no param named `_to`, because that param is sometimes the `_spender`, at other times the `_owner`. That issue predated #9290, however, and the purpose of that PR was to get rid of the `abi-decoder` dependency while preserving existing functionality, so I didn't bother investigating further at the time, and I am no more knowledgeable about it today.

Reply:
removed this change and updated test in https://github.com/MetaMask/metamask-extension/pull/10965/commits/304531b840ba6f0a119455d936afca38e1d2debf